vendor/github.com/OpenBazaar/multiwallet/service/wallet_service.go

```diff
@@ -522,7 +522,7 @@ func (ws *WalletService) saveSingleTxToDB(u model.Transaction, chainHeight int32
 	cb.Value = value
 	cb.WatchOnly = (hits == 0)
 	saved, err := ws.db.Txns().Get(*txHash)
-	if err != nil {
+	if err != nil || saved.WatchOnly != cb.WatchOnly {
```

Comment:
If `err != nil` then `saved` should not be expected to be valid. Did you mean `err != nil && ...`?

Reply:
Answered on slack, reposting here for the records.

"I don't believe so, "ws.db.Txns().Get(*txHash)" returns an err on being unable to get the tx/saved variable for any reason.

In the first scenario where the tx doesn't exist, it would hit the first check in the or statement and continue. 

In the second scenario, where the tx exists and the WatchOnly parameter is wrong, the err would be nil but "saved" would exist. So it should check the second case in the or statement and then continue."

Reply:
Yes, you're right. My mistake. 🤝